Queue.rb

```diff
@@ -1,9 +1,55 @@
+require 'stack'
 class Queue
-  require 'Stack'
-  def initialize
-    stack_in = Stack.new
-    stack_out = Stack.new
-  end
-  def add(element)
-  end
+	def initialize
+		@stack_in = Stack.new
+		@stack_out = Stack.new
+	end
+
+	def empty?
+		return true if @stack_in.is_empty?
+		false
+	end
+	def add(data)
+		@stack_in.push(data)
+	end
+
+	def size
+		counter = 0
+		until @stack_in.is_empty?
+			@stack_out.push(@stack_in.peek)
+			@stack_in.pop
+			counter += 1
+		end
+
+		until @stack_out.is_empty?
+			@stack_in.push(@stack_out.peek)
+			@stack_out.pop
+		end
+		return counter
+	end
+
+	def find_head
+		until @stack_in.is_empty?
+			@stack_out.push(@stack_in.peek)
```

Comment:
same as above. you dont want to lose the none since it can contain more than just data. Like a string and boolean. Thats why you want to add the whole node.